### ISSUE TITLE
Idea for tool: resegment into well-formed sentences

### ISSUE THREAD

**Issue by bricksdont** [cdown/srt#87]
Hi, this is a great library!

I added one more tool in my fork that does automatic sentence segmentation: https://github.com/bricksdont/srt/pull/1

It changes the distribution of subtitle segments so that each subtitle is exactly one well-formed (and complete) sentence. It's not perfect, a machine learning model is involved.

Here is an example:

```srt
# Input

10:01:23,880 --> 10:01:27,640
Regelmässig nimmt er an Veranstaltungen
von FRAGILE Suisse teil,

23
10:01:27,720 --> 10:01:31,840
der Patientenorganisation
für Menschen mit Hirnverletzungen.

# Output

10:01:23,880 --> 10:01:31,840
Regelmässig nimmt er an Veranstaltungen
von FRAGILE Suisse teil, der Patientenorganisation
für Menschen mit Hirnverletzungen.
```

Would you be interested in a PR for this?

**Comment by cdown** [comment]
This looks wonderful and would make a great addition to the repository as part of `srt_tools`! My only point of note is that the machine learning part would need to be an optional dep assuming it's heavy, but that's it :)

**Comment by bricksdont** [comment]
What would be your preferred way of making this an optional dependency? Just letting the user run into an import error? (yes it's heavy :-))

**Comment by cdown** [comment]
I guess wrap the ImportError and provide some nice message, but yes. There's also the question of ongoing maintenance -- are you happy to help keep it up to date with new Python versions, for example?

I suppose this should probably go in `srt_tools/contrib` since it's not under the same maintenance conditions as the rest of the repository.

**Comment by cdown** [comment]
Oh, and can I also take a look at the current code before committing to anything :-)

**Comment by bricksdont** [comment]
> Oh, and can I also take a look at the current code before committing to anything :-)

Yes, of course. Any feedback is welcome, and of course you are not obliged to merge this code.

Here is a Colab that shows basic usage: https://colab.research.google.com/drive/1OHBylPv-8s__IU9_lwTW5CLwHQvfB9Rt?usp=sharing

**Comment by nicorikken** [comment]
I'm looking for a similar feature. I found [srt_equalizer](https://github.com/peterk/srt_equalizer/) by @peterk as a possible solution, which builds on srt. I thought that this would be so fundamental that it should be included in the srt package and then found this issue. Rather than machine learning, perhaps there are certain heuristics that can be used in splitting: don't split a short sentence, don't include only one or two words of a following next sentence, prefer breaking at punctuation, respect non-break-space. For my use-case I need it to support international languages which comes with different punctuation conventions and would be right-to-left as well.

**Comment by bricksdont** [comment]
@nicorikken my solution only uses spacy, which I would not describe as heavy ML. For my use case (machine translation) simple heuristics were not enough. But are welcome to propose a different, heuristic solution and I would test it against mine.

(it also seems this library is not very actively developed in recent years)
